### DISCUSSION TITLE
web automation/scrapping.

### DISCUSSION THREAD

**Discussion by Khalilsqu** [pockethost/pockethost#392] [Q&A] [answered]
Can I do any kind web/browser automation tasks (clicking buttons etc.) when extending with JavaScript (pb_hooks)? Is it possible to use a tool like puppeteer?

**Comment by benallfree** [comment, marked as answer]
PocketBase only supports pure EcmaScript. Puppeteer relies upon a nodejs environment and therefore will not work as a PB hook.
